mesonbuild/dependencies/dev.py

```diff
@@ -530,6 +547,40 @@ def __init__(self, environment: 'Environment', kwargs: T.Dict[str, T.Any]):
         java_home_include = self.java_home / 'include'
         self.compile_args.append(f'-I{java_home_include}')
         self.compile_args.append(f'-I{java_home_include / platform_include_dir}')
+
+        if modules:
+            java_home_lib = self.java_home / 'lib'
+
+            # Location of libraries is different in JDK < 9. Cannot confirm if
+            # this logic works for anything < 1.8, but might as well try.
+            if version_compare(self.version, '<= 1.8'):
+                # The JDK and Meson have a disagreement here, so translate it
+                # over. In the event more translation needs to be done, add to
+                # following dict.
+                def cpu_translate(cpu: str) -> str:
+                    java_cpus = {
+                        'x86_64': 'amd64'
+                    }
+
+                    return java_cpus.get(cpu, cpu)
+
+                if len(modules) > 0:
+                    self.link_args.append(f'-L{java_home_lib / "jre" / cpu_translate(m.cpu_family)}')
+
+                if 'jvm' in modules:
+                    self.link_args.append(f'-ljvm')
+
+                if 'awt' in modules:
+                    self.link_args.append(f'-ljawt')
+            else:
+                if 'jvm' in modules:
+                    self.link_args.append(f'-L{java_home_lib / "server"}')
+                    self.link_args.append(f'-ljvm')
+
+                if 'awt' in modules:
+                    self.link_args.append(f'-L{java_home_lib}')
+                    self.link_args.append(f'-ljawt')
```

Comment:
Maybe these should be `extend([f'-L{java_home_lib}', '-ljawt'])` etc.?

Also you do not need f-strings for string literals so please remove those. It's a lint error (but we don't run flake8 in CI yet... one of these days I will get there...)

Reply:
Mistake! Extend API sounds good.